### PR TITLE
fix(docker source): do not emit component error for out of order logs

### PR DESCRIPTION
**Pull request by dsmith3197**
Fixes https://github.com/vectordotdev/vector/issues/17747

We mistakenly began treating this as an error in https://github.com/vectordotdev/vector/pull/14449

### CODE REVIEW

File: src/sources/docker_logs/mod.rs [hunk 1]
```diff
@@ -994,25 +993,38 @@ impl ContainerLogInfo {
...
-                            container_id: self.id.as_str(),
-                            timestamp_str,
-                        });
-                        return None;
+                    None => {
+                        if self.created < timestamp.with_timezone(&Utc) {
+                            // Noop - first log to process.
+                        } else {
+                            // Received a log with a timestamp before that provided to the Docker API.
+                            // This should not happen, but if it does, we can just ignore these logs.
```
Comment (StephenWakely) [inline src/sources/docker_logs/mod.rs:1020]: I'd argue that if this should not happen - but somehow it does, then we should be raising an error.

**Review by StephenWakely — APPROVED** [review]
I wasn't aware of the container generation changing producing duplicate logs when I made that change... :-/

**Comment by dsmith3197** [comment]
> Does this actually fix the original issue? I think it's just changing the emitted logs?

I don't think there is an actual "issue" that needs to be resolved here. Rather, the correct behavior is to ignore the logs in these particular scenarios, as we have already processed them. These are edge cases that are a consequence of the Docker API, and should not be treated as errors in my opinion.

File: src/sources/docker_logs/mod.rs (hunk 1 above)
Comment (dsmith3197) [reply src/sources/docker_logs/mod.rs:1020]: It *shouldn't* happen from my understanding of the Docker API. However, in the case that this does occur, I would argue that it is not an error that we need to surface to the user.  The Docker Logs source states that it only processes logs with timestamps after the time that it was started. As such, it should ignore these logs by design.

**Pull request merged**